src/rust/engine/graph/src/entry.rs

```diff
@@ -731,6 +731,18 @@ impl<N: Node> Entry<N> {
   ///   acquiring the graph lock here, which we currently don't do.
   ///
   pub(crate) fn clear(&mut self, graph_still_contains_edges: bool) {
+    self.reset_state(if graph_still_contains_edges {
+      ResetStateAction::DirtyPrevious
+    } else {
+      ResetStateAction::JustReset
+    })
+  }
```

Comment:
I like your use of an enum here: boolean parameters are in general not a good idea, so I would actually expose the enum as the parameter of `fn clear` instead, and then remove `fn clear_mem`.

Reply:
`pub(crate)` means that it is only public to consumers _inside_ the crate. Bare `pub` means outside.

Reply:
Heh okay i interpreted that exactly wrong